### PR TITLE
Add form

### PR DESCRIPTION
**Pull request by the-faizmohammad**
**Pull Request Summary**

**Implemented Changes:**

1. **Test Units Enhancement:**
   - Updated test units for index and show action methods to align with recent milestone updates.

2. **User Management:**
   - Introduced a `current_user` method in the ApplicationController, providing the first user from the Users database.

3. **Route Refinement:**
   - Optimized application routes by employing nested resources for comments and likes, specifying actions such as index, new, create, and show.

4. **New Post View:**
   - Added a `new.html.erb` view for posts' new action, streamlining the process of adding a new post to the user.

5. **Index View Enhancement:**
   - Modified the Index view of posts to render the new post action, establishing a route to `users/:user_id/posts/new`.

6. **PostsController Updates:**
   - Enhanced the `PostsController` with new and create actions, facilitating the creation of post instances, rendering posts to the first user, and updating the database.

7. **CommentsController Implementation:**
   - Implemented the `CommentsController` to enable the addition of new comments to posts. The structure mirrors the process of adding new posts, with updated new and create action methods.

8. **New Comment View:**
   - Introduced a new view page rendering a form for adding new comments. The route is defined as `/posts/:post_id/comments/new`.

9. **LikesController Creation:**
   - Created `LikesController` to handle the addition of likes to posts. The controller features a create action for creating instances and updating the database.
   
   
image

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    @like = Like.new(post_id: params[:post_id], user_id: params[:user_id])
+    @like.user = current_user
+    if @like.save
+      flash[:success] = 'Liked the Post!'
+      redirect_to user_post_path(id: @like.post_id, user_id: @like.user_id)
+    else
+      flash.now[:error] = 'Like to the Post could not be saved!'
+    end
+  end
+end
```
Comment (danmainah) [inline app/controllers/comments_controller.rb]: - [ ] @the-faizmohammad Kindly consider making this controller a `CommentsController` with action methods that enable users to add new comments, currently, I am not able to add new comments in your app.

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @the-faizmohammad , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ Added the current_user method in the ApplicationController to make current user data available to all controllers.
✔️ Added action methods to create posts, and likes
✔️ Added forms to add new posts, comments, and a button to add likes to a post.


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    @comment = Comment.new(comment_params)
+    @comment.user = current_user
+    if @comment.save
+      flash[:success] = 'Comment added successfully!'
+      redirect_to user_post_path(id: @comment.post_id, user_id: @comment.user_id)
+    else
+      flash.now[:error] = 'Error: Comment could not be saved!'
+      render :new, locals: { comment: @comment }
+    end
+  end
```
Comment (Olamarx) [inline app/controllers/comments_controller.rb:17]: - [ ] I will encourage you to check the video I made [here](https://www.loom.com/share/6509a44b6fc5423eac04ad0f14ca072c?sid=77d03c3a-f8e6-46de-ad08-4749b6776175) where I explained the issue visually. It is about the comment, it adds it to only the first user, not to each user. Kindly take a double look.

**Review by Olamarx — CHANGES_REQUESTED** [review]
Hi @the-faizmohammad, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- Linters passed ✔️
- Comment feature added ✔️
- Gitflow ✔️
 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by OyePriscilla — APPROVED** [review]
Hi @the-faizmohammad 👍 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights
Adjusted post controller method ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**